### ISSUE TITLE
pre_release() should be smart about version numbers

### ISSUE THREAD

**Issue by krlmlr** [cynkra/fledge#401]
0.x.99.9yyy -> 0.(x+1).0
x.99.99.9yyy -> (x+1).0.0

We also might want new modes for `bump_version()`: `"pre-minor"` -> x.y.99.9000, `"pre-major"` -> `x.99.99.9000`

This is likely a duplicate.

**Comment by maelle** [comment]
It's related to #341 (but not a duplicate of it).

**Comment by krlmlr** [comment]
Yeah, let's do the manual route first. It seems useful for automation too.

**Comment by krlmlr** [comment]
If we keep that automated use case in mind during development -- my vision is that bumping the `"dev"` version would use `"pre-minor"` or `"pre-major"` if there is a feature or breaking change in the list.

**Comment by maelle** [comment]
As the smart behavior will depend on the version decomposition that will be introduced in #412 (here we will need the components to be numeric)

> 0.x.99.9yyy -> 0.(x+1). 
x.99.99.9yyy -> (x+1).0.0

Meaning (in pseudo code), in a function called from pre-release that would return the which **if not provided by the user**,

```
# could the dev component simply exist, does it need to start with 9
if !(components[dev] starts with 9 && components[patch]==99) {
  # normal behavior
  "patch"
  # or error? probably not error
}
if (components[major] == 99 {
  "major"
} else {
 "minor"
}
```

**Comment by krlmlr** [comment]
Perhaps:

if minor == 99: major
else if patch == 99: minor
else: patch

**Comment by maelle** [comment]
ah yes that is simpler. :smile_cat:

**Comment by maelle** [comment]
I'll get to this after #412 is merged.

**Comment by maelle** [comment]
Does "next" we added in #524 cover this?

**Comment by krlmlr** [comment]
Yes, thanks.
